src/misc/m.tnm.download/m.tnm.download.py

```diff
@@ -218,10 +219,13 @@ def download_file(item, code, compare_file_size):
         )
 
     grass.message(_("Downloading %s for %s...") % (filename, name))
-    with open(filename, "wb") as f:
-        for chunk in res.iter_content(chunk_size=1024):
-            if chunk:
-                f.write(chunk)
+    try:
+        with urlopen(url) as inf, open(filename, "wb") as outf:
+            outf.write(inf.read())
+    except urllib.error.HTTPError as e:
+        grass.warning(
+            _("Failed to download %s with status code %d") % (filename, e.code)
```

Comment:
`.format(filename=..., code=...)` is preferred over C-like `%s %d`. The new code does not have to be outdated just because the surrounding code is.

Reply:
I thought C-like formats would be better for cross-language translations, but anyway, we already have Python formats, so not a big deal.

Reply:
We definitively use `.format` in new Python code. So far, we weighted the clarity of code and possible order changes in translations over translation reuse. Plus these are not really translated at this point, no? So it is only the code.

Reply:
@wenzeslaus Shouldn't we prefer f-strings (better readability and shorter) in that case when the minimum Python version we support is 3.8?

Reply:
We didn't adapt the CI yet, but for the main branch, the minimum supported version agreed upon is 3.9. So you are free

Reply:
I thought this was this GRASS repo, not the GRASS addons repo. For addons I don't know, but hope to follow the main grass repo to allow sharing similar maintenance

Reply:
The rule now is: f-strings for plain strings, but `_("...").format(key=...)` for translatable ones. (f-strings are overall better and recommended by linters but don't work with translations.)

Reply:
(https://github.com/OSGeo/grass/blob/main/doc/development/style_guide.md#developing-python-scripts)